editor/backend/src/EditBuiltInCommand.ts

```diff
@@ -245,3 +245,267 @@ export class BasicManipulationCommand extends EditCommand implements BasicManipu
+type ClosestFaceFunction = (info: SubEntityLocationProps) => void;
+
+interface SubEntityGeometryRequestProps {
+  /** Sub-entity to return geometry for */
+  subEntity: SubEntityProps;
+  /** Callback for result */
+  onResult: SubEntityGeometryFunction;
+}
+
+interface ClosestFaceRequestProps {
+  /** Space point */
+  point: XYZProps;
+  /** Optional direction for choosing face from edge or vertex hit... */
+  direction?: XYZProps;
+  /** Callback for result */
+  onResult: ClosestFaceFunction;
+}
+
+interface ElementGeometryCacheOperationRequestProps {
+  /** Target element id, tool element ids supplied by operations between elements... */
+  id: Id64String;
+  /** Callback for result when element's geometry stream is requested in flatbuffer or graphic formats */
+  onGeometry?: ElementGeometryFunction;
+  onSubEntityGeometry?: SubEntityGeometryRequestProps;
+  onClosestFace?: ClosestFaceRequestProps;
+  onOffsetFaces?: OffsetFacesProps;
+}
+
+/** @alpha */
+export class SolidModelingCommand extends BasicManipulationCommand implements SolidModelingCommandIpc {
+  public static override commandId = editorBuiltInCmdIds.cmdSolidModeling;
+
+  private async updateElementGeometryCache(props: ElementGeometryCacheRequestProps): Promise<ElementGeometryCacheResponseProps> {
+    const requestContext = new BackendRequestContext();
+    requestContext.enter();
+    return new Promise<ElementGeometryCacheResponseProps>((resolve, reject) => {
+      if (!this.iModel.isOpen) {
+        reject(new Error("not open"));
+      } else {
+        this.iModel.nativeDb.updateElementGeometryCache(JSON.stringify(props), (ret: IModelJsNative.ErrorStatusOrResult<IModelStatus, ElementGeometryCacheResponseProps>) => {
+          if (ret.error !== undefined)
+            reject(new Error(ret.error.message));
+          else
+            resolve(ret.result!);
+        });
+      }
+    });
+  }
+
+  public async createElementGeometryCache(id: Id64String, filter?: ElementGeometryCacheFilter): Promise<boolean> {
+    const result = await this.updateElementGeometryCache({ id });
+    if (BentleyStatus.SUCCESS !== result.status)
+      return false;
+
+    if (undefined === filter)
+      return true;
+
+    if (filter.minGeom && (undefined === result.numGeom || filter.minGeom > result.numGeom))
+      return false;
+
+    if (filter.maxGeom && (undefined === result.numGeom || filter.maxGeom < result.numGeom))
+      return false;
+
+    if (!filter.parts && (result.numPart ?? 0 > 0))
+      return false;
+
+    if (!filter.curves && (result.numCurve ?? 0 > 0))
+      return false;
+
+    if (!filter.surfaces && (result.numSurface ?? 0 > 0))
+      return false;
+
+    if (!filter.solids && (result.numSolid ?? 0 > 0))
+      return false;
+
+    if (!filter.other && (result.numOther ?? 0 > 0))
+      return false;
+
+    return true;
+  }
+
+  public async clearElementGeometryCache(): Promise<void> {
+    await this.updateElementGeometryCache({});

```

Comment:
I think it's clearer if you use return rather than await

Reply:
Can't return updateElementGeometryCache doesn't return  Promise<void>...

Reply:
then remove the await

Reply:
Removing `await` will produce a floating promise. @bbastings wants `clearElementGeometryCache` to resolve after `updateElementGeometryCache` completes. His code looks exactly correct to me.

Reply:
OK, if `updateGeometryCache` returns a promise (Brien said it didn't - I guess it returns some other type of promise?) then what he has is fine. If it had returned a `Promise<void>` then return rather than await is preferable. 

I bring all this up because i expect lots of copying of this code and i want to start things off right,.